tensorflow/compiler/tf2tensorrt/convert/ops/fill_ops.cc

```diff
@@ -93,43 +142,98 @@ class ConvertFill : public OpConverterBase<ConvertFill> {
+    }
+    const auto &inputs = params.inputs;
+    if (inputs.size() != 1) {
+      return errors::InvalidArgument(
+        op_name, " expects 1 input, but received ", inputs.size());
+    }
+    return Status::OK();
+  }
+
+  Status Convert() {
+    const auto &params = *this->params_;
+    const auto& inputs = params.inputs;
+    auto *converter = params.converter;
+    auto *network = converter->network();
+
+    const TRT_TensorOrWeights& input = inputs.at(0);
+
+    nvinfer1::Dims dims{0};
+
+    nvinfer1::DataType value_type = input.TrtDType();
+
+    std::vector<int> value_input_dims_data = {1};
+    DimsAdapter value_input_dims(value_input_dims_data);
+    StatusOr<TRT_ShapedWeights> value_weights =
+      params.weight_store->GetTempWeights(value_type, value_input_dims);
+    TF_RETURN_IF_ERROR(value_weights.status());
+    TF_RETURN_IF_ERROR(value_weights->SetValues(V));
+    TRT_TensorOrWeights value_input(value_weights.ValueOrDie());
+
+    int is_dims_static = true;
+    ITensorProxyPtr dims_input_tensor;
+    if (!HasStaticShape(input.GetTrtDims())) {
+      is_dims_static = false;
+      auto builder = TRTNetworkBuilder::Create(network,
+                                               params.weight_store);
+      StatusOr<nvinfer1::IShapeLayer*> shape_layer =
+        builder->Shape(input.tensor()->trt_tensor());
+      TF_RETURN_IF_ERROR(shape_layer.status());
+      dims_input_tensor = (*shape_layer)->getOutput(0);
     } else {
-      TF_RETURN_IF_ERROR(PrepareTensorForShape(
-          converter, value_input, scalar_dims, params.validation_only,
-          &scalar_tensor, node_def));
+      dims = input.GetTrtDims();
     }
 
-    auto builder = TRTNetworkBuilder::Create(network, params.weight_store);
-    nvinfer1::Dims beta_shape{1, {nbDims}};
-    StatusOr<nvinfer1::IConstantLayer*> const_layer =
-        builder->Constant(0, beta_shape, value_type);
-    TF_RETURN_IF_ERROR(const_layer.status());
-    ITensorProxyPtr empty_beta_tensor = (*const_layer)->getOutput(0);
+    TRT_TensorOrWeights dims_input(dims_input_tensor);
+    return this->AddFill(value_input,
+                         dims_input,
+                         true,
+                         is_dims_static,
+                         input.GetTrtDims().nbDims,
+                         dims);
 
-    nvinfer1::IFillLayer* layer =
-        network->addFill(trt_dims, nvinfer1::FillOperation::kLINSPACE);
-    TFTRT_RETURN_ERROR_IF_NULLPTR(layer, node_def.name());
-    if (!is_dims_static) {
-      layer->setInput(0, *dims_input.tensor()->trt_tensor());
-    }
-    layer->setInput(1, *scalar_tensor->trt_tensor());
-    layer->setInput(2, *empty_beta_tensor->trt_tensor());
-    converter->SetLayerName(layer, node_def, "fill");
-    ITensorProxyPtr output_tensor = layer->getOutput(0);
-    AddOutput(TRT_TensorOrWeights(output_tensor));
     return Status::OK();
   }
 };
 
 REGISTER_DEFAULT_TRT_OP_CONVERTER(MakeConverterFunction<ConvertFill>(), "Fill");
```

Comment:
why would we want to put this with Fill op in the same file?

Reply:
Because they share the same core implementation, I thought it would make sense... should I revert it back?

Reply:
Do you mean in they share the same core implementation in TensorRT or in the TF-TRT converter? I think you mean the first?
Can we put the new ops to like_ops.cc? If not, we need to rename fill_ops.cc to reflect the fact that it contains multiple ops.

Reply:
@bixia1, as we have discussed offline, the converters are in the same file because they share the same base class. (Note that @drivanov is also working on a converter for `tf.range`, which will share the same base class.) But once `AddFill` is moved to layer_utils, then the base class will be empty, and it could be removed. That means it is easy to put the converters into separate file. I am still slightly inclined to keep them together as related converters.